src/many-ledger/tests/common/mod.rs

```diff
@@ -205,19 +218,53 @@ impl Setup {
         (info.height, r)
     }
 
+    pub fn add_roles_as(
+        &mut self,
+        id: Identity,
+        account_id: Identity,
+        roles: BTreeMap<Identity, BTreeSet<account::Role>>,
```

Comment:
Nit: If you put `IntoIterator<...>` you can use `[...]` instead of having to build a BTreeMap.